src/Components/Server/src/DependencyInjection/ConfigureStaticFilesOptions.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Components.Server.DependencyInjection
+{
+    public class ConfigureStaticFilesOptions : IPostConfigureOptions<StaticFileOptions>
+    {
+        public ConfigureStaticFilesOptions(IWebHostEnvironment environment)
+        {
+            Environment = environment;
+        }
+
+        public IWebHostEnvironment Environment { get; }
+
+        public void PostConfigure(string name, StaticFileOptions options)
+        {
+            name = name ?? throw new ArgumentNullException(nameof(name));
+            options = options ?? throw new ArgumentNullException(nameof(options));
+
+            if (name != Options.DefaultName)
+            {
+                return;
+            }
+
+            // Basic initialization in case the options weren't initialized by any other component
+            options.ContentTypeProvider = options.ContentTypeProvider ?? new FileExtensionContentTypeProvider();
+            if (options.FileProvider == null && Environment.WebRootFileProvider == null)
+            {
+                throw new InvalidOperationException("Missing FileProvider.");
+            }
+
+            options.FileProvider = options.FileProvider ?? Environment.WebRootFileProvider;
+
+            var prepareResponse = options.OnPrepareResponse;
+            if (prepareResponse == null)
+            {
+                options.OnPrepareResponse = BlazorApplicationBuilderExtensions.SetCacheHeaders;
```

Comment:
hmmm

Reply:
This is one of the first things I've seen (based on this approach) which makes me wonder if it's not totally generic.

Reply:
IMO, it's totally generic, but IMO in general it's easy for us to get it right, but easy for users to get wrong. It also makes it really hard to reorganize individual callbacks,etc in order to resolve conflicts (what if you have two pieces that add a callback and override each other).

In general, when we have this type of composition, I like it when it's backed by primitives and visible, so that:
1) I get a clear understanding of how the system works.
2) I can reorder/reconfigure the system to suit my needs without having to go bonkers or digging into the details of how each component on the stack works.

Reply:
If we extend the static files system to have a more native mechanism for plugging in multiple content sources, perhaps it also needs to be able to have things like response header defaults on a per-content-source basis.

Reply:
Yeah Steve found a better way of saying what I was thinking. It's a bit of a heavy hand for us configure caching settings for all static files, yet we do care about having this set for our stuff.

Reply:
I think it's OK to have their quirk in this release, but we should *known issues* or *release notes* this.